### DISCUSSION TITLE
HttpClient.Timeout error

### DISCUSSION THREAD

**Discussion by Drosos-kalyvas** [betalgo/openai#507] [Q&A] [answered]
Hello,
I've been having a problem with client timeout, more specifically I am sending a request that takes a lot of time to process due amount of data and I get a client timeout error after 100 seconds. Any thoughts/ideas on how to change the timeout limit?
This is my code calling the api:

```
var chatResult = await chat.ChatCompletion.CreateCompletion(new ChatCompletionCreateRequest
{
    Messages = new List<ChatMessage>
    {
        ChatMessage.FromSystem(chatRoleFunc),
        ChatMessage.FromUser(prompt)
    },
    Temperature = (float)0.1,
    Tools = new List<ToolDefinition> { newRequestFunction },
     
});
```

and this is the error message:
![error](https://github.com/betalgo/openai/assets/152853608/afd71687-e69d-47e9-b057-6c71c0c99c3b)

It only happens on this specific request out of ~150 I've tried it with, so otherwise it works fine.
Thanks in advance!

**Comment by kayhantolga** [comment, marked as answer]
Depending on how you initialize the library, you can set a timeout. Here's an example:
```
serviceCollection.AddOpenAIService()
    .ConfigureHttpClient(client =>
    {
        client.Timeout = TimeSpan.FromSeconds(180); // Sets the timeout to 180 seconds
    }); 
```

**Reply by Drosos-kalyvas** [reply]
Great, thank you for your reply!
